### PR TITLE
chore(docs): add internals doc

### PR DESCRIPTION
**Pull request by emjburns**
I tried to capture both code and how to think about Swabbie.

### CODE REVIEW

File: INTERNALS.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+### Dry Run
+
+This will ensure swabbie runs in dryRun, meaning no writes, nor any destructive action of the data will occur
+```
+swabbie:
+  dryRun: true
+```
+It's also possible to turn on dryRun at a resource type level
+
+### Configuration
```
Comment (robzienert) [inline INTERNALS.md]: I don't think `Configuration` belongs in the `INTERNALS.md` -- this should continue to live in `README.md` imo.

File: INTERNALS.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+**Account**: a cloud provider's way of grouping resources
+
+**Namespace**: a [SwabbieNamespace](swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/model/SwabbieNamespace.kt) that represents a group of ResourceType in a cloud provider/account/region.
+
+### What does Swabbie look at?
+
+Swabbie was build to clean up things that Spinnaker knows about. 
+Some examples of those types of resources are Images, Snapshots, Autoscaling Groups, Load balancers, and Firewall Rules.
+Identifying that these resources exist is easy.
```
Comment (robzienert) [inline INTERNALS.md]:
```suggestion
Identifying that these resources exist is easy;```

Comment (robzienert) [reply INTERNALS.md]: This didn't really come through... but I'd merge these two sentences with a semi-colon.

File: INTERNALS.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+### What problem is Swabbie solving?
+
+When you're creating resources it is easy to forget about them. 
+This creates cruft in an account which costs money and slows down systems that index that account.
+Swabbie cleans up unused resources.
+
+### How do we determine "use"?
+
+Determining use depends on the resource type.
+This determination is most of the custom logic for each resource type (for example, AmazonImageHandler).
```
Comment (robzienert) [inline INTERNALS.md]: Ticks on `AmazonImageHandler`.

File: INTERNALS.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+It is up to the implementation to whether they'd like to do lots of expensive checking, record statistics over time, or be more lenient based on the specific resource.
+
+### Should Swabbie work 24/7?
+
+That's a personal choice. 
+Swabbie has settings to allow it to only work on weekdays during work hours to further minimize the risk of an outage caused by something unexpectedly being deleted during off hours.
+
+### Lifecycle of a Resource
+
+Mark -> (optionally) Notify -> Delete
```
Comment (robzienert) [inline INTERNALS.md]: Ticks on this.

File: INTERNALS.md (hunk 1 above)
Comment (emjburns) [reply INTERNALS.md]: Ok, I'll move it.

File: INTERNALS.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+**Work Configuration**: The configuration for a distinct cloud provider + resource + regions + settings
+
+**Account**: a cloud provider's way of grouping resources
+
+**Namespace**: a [SwabbieNamespace](swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/model/SwabbieNamespace.kt) that represents a group of ResourceType in a cloud provider/account/region.
+
+### What does Swabbie look at?
+
+Swabbie was build to clean up things that Spinnaker knows about.
```
Comment (jeyrschabu) [inline INTERNALS.md]: s/build/built/

File: INTERNALS.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+A resource is _marked_ when the handler has determined it is not in use.
+As a resource is marked the handler also determines ownership (a DL, usually).
+
+Right after being marked, the resource is _notified_ via the ownership information.
+The notification contains an opt out link which, if clicked, will prevent a resource from ever being cleaned up.
+You can configure a resource to skip notification (if no one would ever opt out or there would be too many notifications).
+
+There's a waiting period between notifying and deleting (configurable) to ensure that an owner has time to opt out.
+
+After this waiting period a resource re-checked to make sure it still qualifies for deletion, then it is deleted.
```
Comment (jeyrschabu) [inline INTERNALS.md]: missing "is re-checked"

**Pull request merged**